src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -176,6 +182,38 @@ def create(cls, catalog: CatalogName) -> 'RepositoryPlugin':
     def sources(self) -> AbstractSet[TDRSourceSpec]:
         return self._sources
 
+    @cached_property
+    def _configured_source_specs_by_name(self) -> Dict[str, TDRSourceSpec]:
```

Comment:
I can see how we would want to constrain the name to be unique
per catalog, but across catalogs, no. I think list_sources
should be parameterized with the catalog. I also don't think
we need to cache this.